strawberry/schema/schema_converter.py

```diff
@@ -0,0 +1,324 @@
+
+    def from_enum(self, enum: EnumDefinition) -> GraphQLEnumType:
+
+        assert enum.name is not None
+
+        # Don't reevaluate known types
+        if enum.name in self.type_map:
+            graphql_enum = self.type_map[enum.name].implementation
+            assert isinstance(graphql_enum, GraphQLEnumType)  # For mypy
+            return graphql_enum
+
+        graphql_enum = GraphQLEnumType(
+            name=enum.name,
+            values={item.name: self.from_enum_value(item) for item in enum.values},
+            description=enum.description,
+        )
+
+        self.type_map[enum.name] = ConcreteType(
+            definition=enum, implementation=graphql_enum
+        )
+
+        return graphql_enum
+
+    def from_enum_value(self, enum_value: EnumValue) -> GraphQLEnumValue:
+        return GraphQLEnumValue(enum_value.value)
+
+    def from_directive(self, directive: DirectiveDefinition) -> GraphQLDirective:
+
+        graphql_arguments = {}
+        for argument in directive.arguments:
+            assert argument.name is not None
+            graphql_arguments[argument.name] = self.from_argument(argument)
+
+        return GraphQLDirective(
+            name=directive.name,
+            locations=directive.locations,
+            args=graphql_arguments,
+            description=directive.description,
+        )
+
+    def from_field(self, field: FieldDefinition) -> GraphQLField:
+
+        field_type = self.get_graphql_type_field(field)
+        field_type = cast(GraphQLOutputType, field_type)
+
+        resolver = self.from_resolver(field)
+        subscribe = None
+
+        if field.is_subscription:
+            subscribe = resolver
+            resolver = lambda event, *_, **__: event  # noqa: E731
+
+        graphql_arguments = {}
+        for argument in field.arguments:
+            assert argument.name is not None
+            graphql_arguments[argument.name] = self.from_argument(argument)
+
+        return GraphQLField(
+            type_=field_type,
+            args=graphql_arguments,
+            resolve=resolver,
+            subscribe=subscribe,
+            description=field.description,
+            deprecation_reason=field.deprecation_reason,
+        )
+
+    def from_input_field(self, field: FieldDefinition) -> GraphQLInputField:
+        if field.default_value in [undefined, UNSET]:
+            default_value = Undefined
+        else:
+            default_value = field.default_value
+
+        field_type = self.get_graphql_type_field(field)
+        field_type = cast(GraphQLInputType, field_type)
+
+        return GraphQLInputField(
+            type_=field_type, default_value=default_value, description=field.description
+        )
+
+    def from_input_object_type(self, object_type: Type) -> GraphQLInputObjectType:
+        type_definition = object_type._type_definition
+
+        return GraphQLInputObjectType(
+            name=type_definition.name,
+            # TODO: Does this need to be deferred?
```

Comment:
good question, I'd keep it deferred as there might be some places where people might have circular dependencies even for input types.